Wayland Debug information (#956)
napari/docs: docs/developers/contributing/testing.md

There are several known issues with displaying GUI tests on windows in CI, and
so certain tests have been disabled from windows in CI, see
[#1377](https://github.com/napari/napari/pull/1377) for more discussion.

### Running napari tests on Wayland 

In napari test suite we use `pyautogui` to simulate mouse and keyboard events.
On Linux it uses `python-xlib` to implement its utility. 

> Unfortunately it do not work ou of the box on Wayland (even with `xwayland` installed)

Author: ```suggestion
Unfortunately it might work ou of the box on Wayland (even with `xwayland` installed). We observed this with GNOME Shell desktop. 
```

```pytb
...
.venv/lib/python3.12/site-packages/Xlib/protocol/display.py:129: in __init__
    raise error.DisplayConnectionError(self.display_name, r.reason)
E   Xlib.error.DisplayConnectionError: Can't connect to display ":0": b'Authorization required, but no authorization protocol specified\n'
```

Based on experiments we found that execution of `xhost +SI:localuser:$USER` solves the problem. 

> It might be configured in autostart by add givent content to `~/.config/autostart/xhost.desktop`

Reviewer: ````suggestion
In the napari test suite we use `pyautogui` to simulate mouse and keyboard events.
On Linux it uses `python-xlib` to implement this utility. 
Unfortunately it might not work out of the box on Wayland (even with `xwayland` installed). We observed this with (for example) the GNOME desktop. 

```pytb
...
.venv/lib/python3.12/site-packages/Xlib/protocol/display.py:129: in __init__
    raise error.DisplayConnectionError(self.display_name, r.reason)
E   Xlib.error.DisplayConnectionError: Can't connect to display ":0": b'Authorization required, but no authorization protocol specified\n'
```

Based on experiments we found that executing `xhost +SI:localuser:$USER` solves the problem. 

It can be done automatically on startup by adding the above line to `~/.config/autostart/xhost.desktop`.
````

```desktop
[Desktop Entry]
Type=Application
Name=Xhost Fix for XWayland
Exec=xhost +SI:localuser:grzegorz
Terminal=false
NoDisplay=true

> ```

Reviewer: I never had to do this, and I'm on wayland :thinking:

Author: You run test with `pytest .` 

I tested this also on fresh VM with Ubuntu. 

After your comment I have set up a fresh Kubuntu VM and there is no such problem here. 

Created VM With Fedora (with gnome) and it is required.

Author: Added information that it happens with GNOME shell.

> You need to add executable property to this file, `chmod +x ~/.config/autostart/xhost.desktop`

Reviewer: ```suggestion
This file also needs to be made executable with `chmod +x ~/.config/autostart/xhost.desktop`.
```
